modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/loading/TargetCheck.scala

```diff
@@ -47,7 +48,15 @@ object TargetCheck {
       Logging[F].debug(show"Caught exception during target check: ${e.toString}")
 
   /** Check if error is worth retrying */
-  def isWorth(e: Throwable): Boolean =
-    e.toString.toLowerCase.contains("(700100) connection timeout expired. details: none") ||
-      e.toString.toLowerCase.contains("(500051) error processing query/statement")
+  def isWorth: Throwable => Boolean = {
+    case e: SQLException => worthRetrying.contains(e.getMessage.toLowerCase)
```

Comment:
* Previous solution checked that given message is present anywhere in the `e.toString` (`e` being `Throwable`)
* New implementation checks takes the exact message of an exception and looks for the match. 

Are these equivalent? If not, is it intentional?

I'm asking because if messages were longer than the ones present in the `worthRetrying` list, like: `(500051) error processing query/statement. Some different part of the message we don't care about...` it would've been caught previously and now not...?

Reply:
This was not intentional.  Good that you spotted it.  I was trying to make the whole thing a bit neater, as we need to add more known messages, but that is a mistake.